ui/appui-react/src/appui-react/pickers/ViewSelector.tsx

```diff
@@ -146,29 +150,30 @@ export class ViewSelector extends React.Component<ViewSelectorProps, ViewSelecto
     this.setState(args, async () => this.loadViews());
   };
 
-  private setStateContainers(views3d: ListItem[], views2d: ListItem[], sheets: ListItem[], unknown?: ListItem[]) {
+  private setStateContainers(views3d: ListItem[], views2d: ListItem[], sheets: ListItem[], unknown?: ListItem[],
+    views3dFiltered?: ListItem[], views2dFiltered?: ListItem[], sheetsFiltered?: ListItem[], unknownFiltered?: ListItem[]) {
     const views3dContainer: ListItem = {
       key: "views3dContainer",
       name: UiFramework.translate("viewTypes.spatialViews"),
       enabled: false,
       type: ListItemType.Container,
-      children: views3d,
+      children: views3dFiltered! ? views3dFiltered : views3d,
```

Comment:
Having the `!` here is not useful, as this code says: `views3dFiltered` is always defined, check if it is defined, then use it, or use something else.